python/lsst/ip/isr/isrFunctions.py

```diff
@@ -444,3 +445,46 @@ def overscanCorrection(ampMaskedImage, overscanImage, fitType='MEDIAN', order=1,
         raise pexExcept.Exception('%s : %s an invalid overscan type' % \
             ("overscanCorrection", fitType))
     ampImage -= offImage
+
+
+def attachTransmissionCurve(exposure, opticsTransmission=None, filterTransmission=None,
+                            sensorTransmission=None, atmosphereTransmission=None):
+    """Attach a TransmissionCurve to an Exposure, given separate curves for different components.
+
+    Parameters
+    ----------
+    exposure: `lsst.afw.image.Exposure`
+        Exposure object to modify by attaching a the product of all given
+        TransmissionCurves in post-assembly trimmed detector coordinates. Must
+        have a valid Detector attached that matches to the detector associated
```

Comment:
There's no check in the below code that things match. So in what sense "must" it match?

Reply:
There's no way for this code to check, but if the caller passes `Detector` for one CCD and a `TransmissionCurve` that's specifically for a different CCD, anyone using the result of this function won't get the right answers.